sys/kern/bus.c

```diff
@@ -105,6 +107,18 @@ bus_space_t *generic_bus_space = &(bus_space_t){
 };
 /* clang-format on */
 
+static int bus_try_attach(device_t *bus, device_t *dev, driver_t *drv,
+                          int *error) {
+  dev->driver = drv;
+  if (device_probe(dev)) {
+    klog("%s detected!", drv->desc);
+    *error = device_attach(dev);
+    if (*error)
+      return 1;
+  }
+  return 0;
+}
```

Comment:
Where are you using `bus` here?

I think that `error` is not needed here and we could unified `error` with return value.

Reply:
```bus``` was unused and has been removed, ```error``` also has been removed from the parameters of the function.